### ISSUE TITLE
Reproducing The Results

### ISSUE THREAD

**Issue by xiangruhuang** [czq142857/BAE-NET#4]
Hi Zhiqin,

I'm trying to reproduce your results on Shapenet Part Segmentation dataset.

I cloned your code and downloaded your data. Then I trained the network under the one-example setting with the following parameters:
```
python main.py --train --L1reg --supervised --iteration 200000 --pretrain_iters 3000 --retrain_iters 4 --dataset 03001627_vox --data_dir ./data/03001627_chair/ --checkpoint_dir checkpoint_1shot_1_3000 --supervision_list ref1.txt --sample_dir samples/03001627_chair  --real_size 32 --points_per_shape 8192
```

Then I used the following code to test for IoU:
```
python main.py --iou --L1reg --supervised --iteration 200000 --pretrain_iters 3000 --retrain_iters 4 --dataset 03001627_vox --data_dir ./data/03001627_chair/ --checkpoint_dir checkpoint_1shot_1_3000 --supervision_list ref1.txt --sample_dir samples/03001627_chair  --real_size 32 --points_per_shape 8192
```

The results shows an average IoU of 59.5%. Although it does not directly corresponds to a number in your paper, it seems deviated from the average IoU reported in figure 9 for the chair category (which looks like in between 75% to 80%).

Can you kindly confirm if I did anything wrong?

Thanks a lot!

Best,
Xiangru

**Comment by czq142857** [comment]
Were you running the code on Windows or Ubuntu?

**Comment by xiangruhuang** [comment]
On Ubuntu 18.04.

**Comment by czq142857** [comment]
That is strange... The IoU of chairs should be around 80%. Could you try the following?

1) run the code again. If it is still bad, change the parameter "--pretrain_iters 3000" to "--pretrain_iters 4000" and try again.

2) There is one issue of tensorflow on Windows 10 that I know of. There is a slight chance that this issue also happens on Ubuntu != 16.04. You could check if this issue causes the problem.

"It has also been tested on Windows 10 but something went wrong. If sigmoid is placed before reduce_max, sigmoid won't be executed in certain cases. The solution is to change the last few layers of the decoder from "linear - sigmoid - reduce_max" to "linear - reduce_max - sigmoid"."

**Comment by xiangruhuang** [comment]
Hi Zhiqin,

Looks like updating pretrain_iters to 4000 according to the first suggestion would make an IoU of 72.5 for chairs. The second suggestion doesn't have much impact.

Best,
Xiangru

**Comment by czq142857** [comment]
Hi Xiangru,

I trained it on my PC with Ubuntu 16.04, CUDA 10.0, and tensorflow-gpu 1.14.0. The results are as expected.

I think it is most likely a compatibility issue, whether it is because of the Ubuntu or the TF versions.

I have attached my training logs. Hopefully, you can use it to debug.

Best,
Zhiqin

```
----------net summary----------
training samples    3746
supervised samples  1
network branch	  4
-------------------------------


2021-03-03 13:26:08.160315: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcublas.so.10.0
2021-03-03 13:26:12.351819: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7
Epoch: [ 0/53] [3745/3746] time: 112.6818, loss: 0.06478665
Epoch: [ 1/53] [3745/3746] time: 162.6793, loss: 0.05468399
Epoch: [ 2/53] [3745/3746] time: 213.4253, loss: 0.04936141
Epoch: [ 3/53] [3745/3746] time: 264.4256, loss: 0.04558803
Epoch: [ 4/53] [3745/3746] time: 315.4815, loss: 0.04178873
Epoch: [ 5/53] [3745/3746] time: 366.4138, loss: 0.03738222
Epoch: [ 6/53] [3745/3746] time: 418.0479, loss: 0.03471198

./data/03001627_chair/
82.4

Epoch: [ 7/53] [3745/3746] time: 472.9192, loss: 0.03258998
Epoch: [ 8/53] [3745/3746] time: 522.7641, loss: 0.03097905
Epoch: [ 9/53] [3745/3746] time: 573.2374, loss: 0.02945868
Epoch: [10/53] [3745/3746] time: 624.4454, loss: 0.02823434
Epoch: [11/53] [3745/3746] time: 674.9891, loss: 0.02699020
Epoch: [12/53] [3745/3746] time: 725.7418, loss: 0.02596593

./data/03001627_chair/
83.5

Epoch: [13/53] [3745/3746] time: 780.5051, loss: 0.02507024
Epoch: [14/53] [3745/3746] time: 831.0364, loss: 0.02413209
Epoch: [15/53] [3745/3746] time: 881.0706, loss: 0.02334402
Epoch: [16/53] [3745/3746] time: 931.4642, loss: 0.02266314
Epoch: [17/53] [3745/3746] time: 981.7456, loss: 0.02199159
Epoch: [18/53] [3745/3746] time: 1031.9453, loss: 0.02172153

./data/03001627_chair/
85.0

Epoch: [19/53] [3745/3746] time: 1087.4918, loss: 0.02095035
Epoch: [20/53] [3745/3746] time: 1138.3679, loss: 0.02054121
Epoch: [21/53] [3745/3746] time: 1188.5138, loss: 0.02020413
Epoch: [22/53] [3745/3746] time: 1238.9833, loss: 0.01972044
Epoch: [23/53] [3745/3746] time: 1290.0454, loss: 0.01988373
Epoch: [24/53] [3745/3746] time: 1340.4190, loss: 0.01915983

./data/03001627_chair/
85.0

Epoch: [25/53] [3745/3746] time: 1395.8159, loss: 0.01872432
Epoch: [26/53] [3745/3746] time: 1446.6791, loss: 0.01840478
Epoch: [27/53] [3745/3746] time: 1497.1986, loss: 0.01819519
Epoch: [28/53] [3745/3746] time: 1547.6780, loss: 0.01788881
Epoch: [29/53] [3745/3746] time: 1598.6845, loss: 0.01756642
Epoch: [30/53] [3745/3746] time: 1649.6594, loss: 0.01738165

./data/03001627_chair/
84.9

Epoch: [31/53] [3745/3746] time: 1704.7231, loss: 0.01712014
Epoch: [32/53] [3745/3746] time: 1754.6605, loss: 0.01691186
Epoch: [33/53] [3745/3746] time: 1805.2623, loss: 0.01672662
Epoch: [34/53] [3745/3746] time: 1855.5725, loss: 0.01659287
Epoch: [35/53] [3745/3746] time: 1905.9218, loss: 0.01628951
Epoch: [36/53] [3745/3746] time: 1956.9503, loss: 0.01610321

./data/03001627_chair/
85.0

Epoch: [37/53] [3745/3746] time: 2011.6046, loss: 0.01607522
Epoch: [38/53] [3745/3746] time: 2061.9965, loss: 0.01588468
Epoch: [39/53] [3745/3746] time: 2112.3858, loss: 0.01574046
Epoch: [40/53] [3745/3746] time: 2163.3644, loss: 0.01571350
Epoch: [41/53] [3745/3746] time: 2214.0094, loss: 0.01546336
Epoch: [42/53] [3745/3746] time: 2264.3942, loss: 0.01532709

./data/03001627_chair/
85.2

Epoch: [43/53] [3745/3746] time: 2319.3959, loss: 0.01511390
Epoch: [44/53] [3745/3746] time: 2369.8606, loss: 0.01513032
Epoch: [45/53] [3745/3746] time: 2420.5773, loss: 0.01497689
Epoch: [46/53] [3745/3746] time: 2471.2117, loss: 0.01484644
Epoch: [47/53] [3745/3746] time: 2522.0582, loss: 0.01468729
Epoch: [48/53] [3745/3746] time: 2572.5479, loss: 0.01468371

./data/03001627_chair/
84.4

Epoch: [49/53] [3745/3746] time: 2627.7345, loss: 0.01447706
Epoch: [50/53] [3745/3746] time: 2678.7986, loss: 0.01442071
Epoch: [51/53] [3745/3746] time: 2730.1854, loss: 0.01424979
Epoch: [52/53] [3745/3746] time: 2780.6880, loss: 0.01419040
Epoch: [53/53] [3745/3746] time: 2831.2103, loss: 0.01413117

./data/03001627_chair/
85.1
```

**Comment by xiangruhuang** [comment]
Hi Zhiqin,

Thanks a lot for the log.

I agree there are some issues with the compatibility. Let me get back to you on this.

Best,
Xiangru
